src/boot/efi/boot.c

```diff
@@ -2286,6 +2328,115 @@ static VOID config_write_entries_to_variable(Config *config) {
+        static const UINT32 sb_vars_opts =
+                EFI_VARIABLE_NON_VOLATILE |
+                EFI_VARIABLE_BOOTSERVICE_ACCESS |
+                EFI_VARIABLE_RUNTIME_ACCESS |
+                EFI_VARIABLE_TIME_BASED_AUTHENTICATED_WRITE_ACCESS |
+                0;
+
+        static struct {
+                const CHAR16 *name;
+                const CHAR16 *filename;
+                const EFI_GUID vendor;
+                CHAR8 *buffer;
+                UINTN size;
+        } sb_vars[] = {
+                { L"db",  L"db.auth",  EFI_IMAGE_SECURITY_DATABASE_VARIABLE, NULL, 0 },
+                { L"KEK", L"KEK.auth", EFI_GLOBAL_VARIABLE, NULL, 0 },
+                { L"PK",  L"PK.auth",  EFI_GLOBAL_VARIABLE, NULL, 0 },
+        };
+
+        EFI_STATUS err = EFI_SUCCESS;
+        _cleanup_(FileHandleClosep) EFI_FILE_HANDLE dir = NULL;
+
+        err = uefi_call_wrapper(root_dir->Open, 5, root_dir, &dir, keys_dir, EFI_FILE_MODE_READ, 0ULL);
+        if (EFI_ERROR(err)) {
+                return err;
+        }
+
+        /* we start by loading ALL the variables from the disk */
+        for (UINTN i = 0; i < ELEMENTSOF(sb_vars); i++) {
+                err = file_read(dir, sb_vars[i].filename, 0, 0, &sb_vars[i].buffer, &sb_vars[i].size);
+                if (EFI_ERROR(err)) {
+                        Print(L"Failed reading %s file.\n", sb_vars[i].filename);
+                        return err;
+                }
+        }
+
+        /* we then write them to the efivars */
+        for (UINTN i = 0; i < ELEMENTSOF(sb_vars); i++) {
+                err = efivar_set_raw(&sb_vars[i].vendor, sb_vars[i].name, sb_vars[i].buffer, sb_vars[i].size, sb_vars_opts);
+                if (EFI_ERROR(err)) {
+                        Print(L"Failed to write %s secure boot variable.\n", sb_vars[i].name);
+                        return err;
+                }
+                FreePool(sb_vars[i].buffer);
+        }
+
+        /* if we loaded the signature keys successfully then we remove all the
+         * signature keys entries from  the menu as the system is now locked down. */
+        config_remove_by_type(config, LOADER_SIGNATURE_KEYS);
+
+        return err;
+}
+
+static EFI_STATUS secure_boot_discover_keys(Config *config, EFI_FILE *root_dir) {
+        EFI_STATUS err = EFI_SUCCESS;
+        _cleanup_(FileHandleClosep) EFI_FILE_HANDLE keys_basedir = NULL;
+
+        if (!setup_mode_enabled())
+                return EFI_SUCCESS;
+
+        /* if auto enrollement is activated (this is the default), we try to
+         * load keys straight from the /loader/keys/auto location. If it fails 
+         * because the directory is not present we keep going silently, otherwise
+         * we log an error and still proceed to add the entries to the menu
+         * (including the auto one if it exists) */
+        if (config->auto_enroll == true) {
+                err = secure_boot_enroll_from(config, root_dir, (CHAR16*) L"\\loader\\keys\\auto");
+
```

Comment:
drop this empty line. we generally try to keep function calls and their immediate error handling together. 

empty liens are great to separate out logical steps, but we consider error handling the same logical step as the function call it is for...